File: experiments/020-PythonVsJava/cells/python-pytest_1-Empty_free/2019-2c-parcial2-masked/20260921-202806-59410/exercise/java/src/main/java/surveycrawler/CrawlerFacingDown.java
package surveycrawler;

public class CrawlerFacingDown extends CrawlerFacing {

    // facing name

    @Override
    public String facingName() {
        return "Down";
    }

    // moving

    @Override
    public void retreat(SurveyCrawler aCrawler) {
        aCrawler.moveUp();
    }

    @Override
    public void advance(SurveyCrawler aCrawler) {
        aCrawler.moveDown();
    }

    // facing

    @Override
    public void turnCounterClockwise(SurveyCrawler aCrawler) {
        aCrawler.faceRight();
    }

    @Override
    public void turnClockwise(SurveyCrawler aCrawler) {
        aCrawler.faceLeft();
    }
}

File: experiments/020-PythonVsJava/cells/python-pytest_1-Empty_free/2019-2c-parcial2-masked/20260921-212215-18237/output/project/src/surveycrawler/crawler_facing_down.py
from surveycrawler.crawler_facing import CrawlerFacing


class CrawlerFacingDown(CrawlerFacing):

    # facing name

    @classmethod
    def facing_name(cls):
        return "Down"

    # moving

    def retreat(self, a_crawler):
        a_crawler.move_up()

    def advance(self, a_crawler):
        a_crawler.move_down()

    # facing

    def turn_counter_clockwise(self, a_crawler):
        a_crawler.face_right()

    def turn_clockwise(self, a_crawler):
        a_crawler.face_left()
